docs: document default snapshot repository in Kibana (#6683) (#6713)
elastic/docs-content: deploy-manage/tools/snapshot-and-restore/_snippets/default-snapshot-repository.md

The default snapshot repository stores all snapshots generated by data stream lifecycle management. The first repository you register in {{kib}} becomes the default automatically, and you can change which repository is the default at any time.

1. Open the actions menu for the repository you want to use as the default.
2. Select **Set as default**.
3. In the **Change default repository?** dialog, select **Change default** to confirm.

> You can also change the default when you edit a repository: turn on the **Set as default repository** switch, then save. If a different repository is already the default, a confirmation dialog appears before the change is applied.

Author: ```suggestion
- From the list of repositories:
   1. Open the actions menu for the repository you want to use as the default.
   2. Select **Set as default**.
   3. In the **Change default repository?** dialog, select **Change default** to confirm.
- When you edit a repository: Turn on the **Set as default repository** option, then save. If a different repository is already the default, a confirmation dialog appears before the change is applied.
```

> The default snapshot repository stores all snapshots generated by data stream lifecycle management. The first repository you register in {{kib}} becomes the default automatically, and you can change which repository is the default at any time.

Reviewer: ```suggestion
The default snapshot repository is a repository that {{es}} features can use when they need to perform snapshot or restore operations without explicitly specifying a repository. Currently, only [data stream lifecycle management (DLM)](/manage-data/lifecycle/data-stream.md) uses it to create searchable snapshots when moving data to the frozen tier.

When you register the first repository in {{kib}}, it becomes the default automatically. You can change the default repository at any time.
```

Reviewer: For your consideration, a simpler alternative could be:

```
The default snapshot repository is used by [data stream lifecycle management (DLM)](/manage-data/lifecycle/data-stream.md) when moving data to the frozen tier.
```

Or something similar. Whatever aligned with https://docs-v3-preview.elastic.dev/elastic/elasticsearch/tree/main/reference/elasticsearch/configuration-reference/snapshot-restore-settings#repositories-default-repository should be ok!

My first proposal prepares the introduction for a future update if more features use the default repository, or if other APIs start working with the default repo instead of with an explicit repo.

Author: I like your first suggestion! Thank you!

> The default snapshot repository stores all snapshots generated by data stream lifecycle management. The first repository you register in {{kib}} becomes the default automatically, and you can change which repository is the default at any time. The default repository is marked with a **Default** badge in the repositories list and the repository details flyout.

Author: ```suggestion
The default snapshot repository stores all snapshots generated by data stream lifecycle management. The first repository you register in {{kib}} becomes the default automatically, and you can change which repository is the default at any time.
```

> To change the default repository from {{kib}}, you need the `cluster:admin/settings/update` [cluster privilege](elasticsearch://reference/elasticsearch/security-privileges.md#privileges-list-cluster). Without it, the **Default** badge is still visible, but the controls for changing the default are hidden. The default repository is stored in the [`repositories.default_repository`](elasticsearch://reference/elasticsearch/configuration-reference/snapshot-restore-settings.md#repositories-default-repository) cluster setting.

Author: ```suggestion
To change the default repository from {{kib}}, you need the `cluster:admin/settings/update` [cluster privilege](elasticsearch://reference/elasticsearch/security-privileges.md#privileges-list-cluster). The default repository is stored in the [`repositories.default_repository`](elasticsearch://reference/elasticsearch/configuration-reference/snapshot-restore-settings.md#repositories-default-repository) cluster setting.
```

Keep the following constraints in mind:

* The default repository cannot be removed. To remove it, first set another repository as the default.
* Read-only repositories cannot be set as the default.
* URL repositories cannot be set as the default.

To change the default repository, use one of the following methods:

* From the list of repositories:
   1. Open the actions menu for the repository you want to use as the default.
   2. Select **Set as default**.
   3. In the **Change default repository?** dialog, select **Change default** to confirm.

> * When you edit a repository: Turn on the **Set as default repository** option, then save. If a different repository is already the default, a confirmation dialog appears before the change is applied.

Author: ```suggestion
* When you edit a repository: Turn on the **Set as default repository** option, then save.
```

The default snapshot repository stores all snapshots generated by data stream lifecycle management. The first repository you register in {{kib}} becomes the default automatically, and you can change which repository is the default at any time.

The default repository is marked with a **Default** badge in the repositories list and the repository details flyout.

> To change the default repository from the **Repositories** list in {{kib}}:

Author: ```suggestion
```

> When you register the first repository in {{kib}}, it becomes the default automatically. You can change the default repository at any time.

Reviewer 2: Just a quick clarification. When user creates first repository (when there are no other repositories) in Kibana it's not really being automatically set as default but we pre-check the `Set as default repository` switch for the user. The user can change it though.

https://github.com/user-attachments/assets/e3e15e0a-3298-44e4-9de5-54b7f3ea4ecf

Author: Yeah I probably oversimplified here, i'll make this more accurate
